src/leo_s3_user.erl

```diff
@@ -105,36 +112,22 @@ create(UserId, Password, HashType) ->
 
 
 %% @private
-create_fun(UserId, Password, HashType) ->
+create_fun(UserId, Password, AccessKeyId, HashType) ->
     CreatedAt = leo_date:now(),
     Digest = hash_and_salt_password(HashType, Password, CreatedAt),
 
     %% insert a user-info
-    case mnesia:write(?USERS_TABLE,
+    ok = mnesia:write(?USERS_TABLE,
                       #?S3_USER{id = UserId,
                                 password = Digest,
                                 created_at = CreatedAt,
-                                updated_at = CreatedAt}, write) of
-        ok ->
-            %% create a key
-            case leo_s3_auth:create_key(UserId) of
-                {ok, Keys} ->
-                    AccessKeyId = leo_misc:get_value(access_key_id, Keys),
-                    case mnesia:write(?USER_CREDENTIAL_TABLE,
-                                      #user_credential{user_id = UserId,
-                                                       access_key_id = AccessKeyId,
-                                                       created_at = CreatedAt}, write) of
-                        ok ->
-                            {ok, Keys};
-                        _ ->
-                            {error, transaction_abort}
-                    end;
-                Error ->
-                    Error
-            end;
-        _ ->
-            {error, transaction_abort}
-    end.
+                                updated_at = CreatedAt}, write),
+    %% create a key
+    ok = mnesia:write(?USER_CREDENTIAL_TABLE,
+                      #user_credential{user_id = UserId,
+                                       access_key_id = AccessKeyId,
+                                       created_at = CreatedAt}, write),
```

Comment:
As I suggested at the previous review comment, I think mnesia:write called inside https://github.com/leo-project/leofs/blob/v1/apps/leo_manager/src/leo_manager_console.erl#L2119-L2121 leo_s3_user:update/1 also should be included in this transaction to make the whole mnesia operation atomic. (The password field should be updated to SecretAccessKey)